cheatsheets/gleam-for-elm-users.md

```diff
@@ -0,0 +1,699 @@
+
+Additionally, underscores can be added to both Ints and Floats for clarity.
+
+```rust
+const oneMillion = 1_000_000
+const twoMillion = 2_000_000.0
+```
+
+### Strings
+
+In both Elm and Gleam all strings support unicode. Gleam uses UTF-8 binaries. Elm compiles to Javascript which uses UTF-16 for its strings.
+
+Both languages use double quotes for strings.
+
+#### Elm
+
+```elm
+"Hellø, world!"
+```
+
+#### Gleam
+
+```rust
+"Hellø, world!"
+```
+
+### Tuples
+
+Tuples are very useful in both Elm and Gleam as they're the only collection data type that allows mixed types in the collection.
+
+#### Elm
+
+In Elm, tuplies are limited to only 2 or 3 entries. It is recommended to use records for more larger numbers of entries.
+
+```elm
+myTuple = ("username", "password", 10)
+(_, password, _) = myTuple
+```
+
+#### Gleam
+
+There is no limit to the number of entries in Gleam tuples.
+
+```rust
+let my_tuple = #("username", "password", 10)
+let #(_, password, _) = my_tuple
+```
+
+### Lists
+
+Both Elm and Gleam support lists. All entries have to be of the same type.
+
+#### Elm
+
+You can pattern match on lists in Elm, but only in case-statements.
+
+```elm
+list = [2, 3, 4]
+list2 = 1 :: list
+```
+
+#### Gleam
+
+The `cons` operator works the same way both for pattern matching and for appending elements to the head of a list, but it uses a different syntax.
+
+```rust
+let list = [2, 3, 4]
+let list = [1, ..list]
+let [1, second_element, ..] = list
+[1.0, ..list] // compile error, type mismatch
+```
+
+### Dicts
+
+Dict in Elm and Map in Gleam have similar properties and purpose.
+
+In Gleam, maps can have keys and values of any type, but all keys must be of the same type in a given map and all values must be of the same type in a given map.
+
+Like Elm, there is no map literal syntax in Gleam, and you cannot pattern match on a map.
+
+#### Elm
+
+```Dict
+import Dict
+
+Dict.fromList [ ("key1", "value1"), ("key2", "value2") ]
+Dict.fromList [ ("key1", "value1"), ("key2", 2) ] -- Compile error
+```
+
+#### Gleam
+
+```rust
+import gleam/map
+
+map.from_list([#("key1", "value1"), #("key2", "value2")])
+map.from_list([#("key1", "value1"), #("key2", 2)]) // Type error!
+```
+
+## Type Aliases
+
+Elm uses type alises to define the layout of records. Gleam uses Custom Types to achieve a similar
+result.
+
+Gleam's custom types allow you to define a collection data type with a fixed number of named fields, and the values in those fields can be of differing types.
+
+#### Elm
+
+```elm
+type alias Person =
+ { name : String
+ , age : Int 
+ }
+
+person = Person "Jake" 35
+name = person.name
+```
+
+#### Gleam
+
+Gleam's custom types can be used in much the same way. At runtime, they have a tuple representation and are compatible with Erlang records.
+
+```rust
+type Person {
+  Person(name: String, age: Int)
+}
+
+let person = Person(name: "Jake", age: 35)
+let name = person.name
+```
```

Comment:
Not sure what is happening here, GitHub's syntax highlighting seems to have broken?